docs/debugger_v2.md

```diff
@@ -0,0 +1,303 @@
+# Debugging TensorFlow 2 Programs Using TensorBoard Debugger V2
+
+> *NOTE*: tf.debugging.experimental.enable_dump_debug_info() is an experimental
+> API and may be subject to breaking changes in the future.
+
+Catastrophic events involving [NaN](https://en.wikipedia.org/wiki/NaN)s can
+sometimes occur during a TensorFlow program, crippling the model training
+processes. The root cause of such events are often obscure, especially for
+models of non-trivial size and complexity. To make it easier to debug this type
+of model bugs, TensorFlow 2.3+ provides a specialized GUI tool in TensorBoard called Debugger V2. Here we demonstrate how to use this tool by working through a real bug involving NaNs in a neural network written in TensorFlow 2 (TF2).
+
+The techniques illustrated in this tutorial are applicable to other types of
+debugging activities such as inspecting runtime tensor shapes in complex
+programs. This document focuses on a NaN bug due to the high frequency at which
+this type of bugs appear in user programs.
```

Comment:
Nit: can shorten this to "This tutorial focuses on NaNs due to their high frequency of occurrence."

Reply:
Done.